fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/DefaultLoanLifecycleStateMachine.java

```diff
@@ -49,6 +49,17 @@ public void transition(final LoanEvent loanEvent, final Loan loan) {
                 // in case of Loan creation, a LoanCreatedBusinessEvent is also raised, no need to send a status change
                 businessEventNotifierService.notifyPostBusinessEvent(new LoanStatusChangedBusinessEvent(loan));
             }
+
+            // set mandatory field states based on new status after the transition
+            switch (newStatus) {
+                case ACTIVE -> {
+                    if (loan.getClosedOnDate() != null) {
```

Comment:
I think we should add the `overpaidOnDate` to be null as well.

Reply:
PR updated